packages/server/src/game/util.ts

```diff
@@ -0,0 +1,199 @@
+import { Position, Unit, Command, Component, Game, BuildingMap, PresenceMap } from '../types'
+import { getHpComponent, getMoveComponent, getAttackerComponent, getHarvesterComponent, getProducerComponent, getBuilderComponent, getVisionComponent, getBuildingComponent } from './components.js'
+import * as V from '../vector.js'
+
+// This code generates an offset position for a given spiral index
+export function spiral(p: Position, i: number, scale: number) {
+    const offsets = [
+        // target
+        [0,0],
+        // layer 1
+        [1,0],
+        [1,1],
+        [0,1],
+        [-1,1],
+        [-1,0],
+        [-1,-1],
+        [0, -1],
+        [1, -1],
+        // layer 2
+        [2,0],
+        [2,1],
+        [2,2],
+        [1,2],
+        [0,2],
+        [-1,2],
+        [-2,2],
+        [-2,1],
+        [-2,0],
+        [-2,-1],
+        [-2,-2],
+        [-1,-2],
+        [0,-2],
+        [1,-2],
+        [2,-2],
+        [2,-1],
+    ];

```

Comment:
not sure if the spiral is even necessary anymore, but i'll keep it for now for formation movement and remove once i have more tests for it